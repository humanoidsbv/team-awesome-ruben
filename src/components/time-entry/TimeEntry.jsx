import React from 'react';
import PropTypes from 'prop-types';

import DeleteIcon from '../../assets/icons/icon-delete.svg';
import styles from './TimeEntry.module.css';

const TimeEntry = ({
  client,
  deleteEntry,
  timeEntryId,
  startTimestamp,
  stopTimestamp
}) => {
  const totalHours =
    (Date.parse(stopTimestamp) - Date.parse(startTimestamp)) / 1000 / 60 / 60;

  const totalHoursString = `${Math.trunc(totalHours)}:${Math.round(
    (totalHours % 1) * 60
  )
    .toString()
    .padStart(2, '0')}`;

  const startTimeString = new Date(startTimestamp).toLocaleTimeString('nl-NL', {
    hour: '2-digit',
    minute: '2-digit'
  });
  const stopTimeString = new Date(stopTimestamp).toLocaleTimeString('nl-NL', {
    hour: '2-digit',
    minute: '2-digit'
  });

  return (
    <div className={styles.timeEntry}>
      <span>{client}</span>
      <button
        className={styles.deleteEntry}
        onClick={() => deleteEntry(timeEntryId)}
        type="button"
      >
        <DeleteIcon className={styles.deleteIcon} />
        Delete
      </button>
      <div className={styles.timeStamps}>
        <span className={styles.hoursWorked}>
          {`${startTimeString} - ${stopTimeString}`}
        </span>
        <span className={styles.hoursDuration}>{totalHoursString}</span>
      </div>
    </div>
  );
};

TimeEntry.propTypes = {
  client: PropTypes.string.isRequired,
  deleteEntry: PropTypes.func.isRequired,
  startTimestamp: PropTypes.string.isRequired,
  stopTimestamp: PropTypes.string.isRequired,
  timeEntryId: PropTypes.number.isRequired
};

export default TimeEntry;
