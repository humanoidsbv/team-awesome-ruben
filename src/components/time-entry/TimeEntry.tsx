import React from 'react';

import { TimeEntryInterface } from '../../redux/time-entries/types';
import DeleteIcon from '../../assets/icons/icon-delete.svg';
import styles from './TimeEntry.module.css';

export interface TimeEntryProps extends TimeEntryInterface {
  deleteEntry: (number) => {};
}

const TimeEntry = ({
  client,
  deleteEntry,
  id,
  startTimestamp,
  stopTimestamp
}: TimeEntryProps): React.ReactElement => {
  const totalHours =
    (Date.parse(stopTimestamp) - Date.parse(startTimestamp)) / 1000 / 60 / 60;

  const totalHoursString = `${Math.trunc(totalHours)}:${Math.round(
    (totalHours % 1) * 60
  )
    .toString()
    .padStart(2, '0')}`;

  const startTimeString = new Date(startTimestamp).toLocaleTimeString('nl-NL', {
    hour: '2-digit',
    minute: '2-digit'
  });
  const stopTimeString = new Date(stopTimestamp).toLocaleTimeString('nl-NL', {
    hour: '2-digit',
    minute: '2-digit'
  });

  return (
    <div className={styles.timeEntry}>
      <span>{client}</span>
      <button
        className={styles.deleteEntry}
        onClick={(): {} => deleteEntry(id)}
        type="button"
      >
        <DeleteIcon className={styles.deleteIcon} />
        Delete
      </button>
      <div className={styles.timeStamps}>
        <span className={styles.hoursWorked}>
          {`${startTimeString} - ${stopTimeString}`}
        </span>
        <span className={styles.hoursDuration}>{totalHoursString}</span>
      </div>
    </div>
  );
};

export default TimeEntry;
