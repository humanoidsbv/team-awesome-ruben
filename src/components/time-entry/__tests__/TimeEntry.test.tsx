import * as React from 'react';
import { shallow } from 'enzyme';

import TimeEntry from '../TimeEntry';

it('converts basic button using shallow', () => {
  const timeEntry = shallow(
    <TimeEntry
      id={123}
      client={{
        branch: '123',
        branchAdress: '123',
        companyName: '123',
        id: 123,
        locality: '123',
        postalCode: '123'
      }}
      deleteEntry={(a: number) => null}
      startTimestamp="2019-10-16T09:00:00.000Z"
      stopTimestamp="2019-10-16T17:30:00.000Z"
    />
  );
  expect(timeEntry).toMatchSnapshot();
});
