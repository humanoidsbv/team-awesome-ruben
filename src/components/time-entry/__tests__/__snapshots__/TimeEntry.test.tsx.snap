// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts basic button using shallow 1`] = `
<div
  className="timeEntry"
>
  <span
    className="textField"
  >
    123
  </span>
  <button
    className="deleteEntry"
    onClick={[Function]}
    type="button"
  >
    <test-file-stub
      className="deleteIcon"
    />
    Delete
  </button>
  <div
    className="timeStamps"
  >
    <span
      className="hoursWorked"
    >
      11:00 - 19:30
    </span>
    <span
      className="hoursDuration"
    >
      8:30
    </span>
  </div>
</div>
`;
